content/en/docs/pipelines/sdk/python-function-components.ipynb

```diff
@@ -0,0 +1,494 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Building Python function-based components\n",
+    "> Building your own lightweight pipelines components using Python\n",
+    "\n",
+    "A Kubeflow Pipelines component is a self-contained set of code that performs one step in your\n",
+    "ML workflow. A pipeline component is composed of:\n",
+    "\n",
+    "*   The component code, which implements the logic needed to perform a step in your ML workflow.\n",
+    "*   A component specification, which defines the following:\n",
+    "    \n",
+    "    *   The component's metadata, its name and description.\n",
+    "    *   The component's interface, the component's inputs and outputs.\n",
+    "    *   The component's implementation, the Docker container image\n",
+    "        to run, how to pass inputs to your component code, and how\n",
+    "        to get the component's outputs.\n",
+    "\n",
+    "Python function-based components make it easier to iterate quickly by letting you build your\n",
+    "component code as a Python function and generating the [component specification][component-spec] for you.\n",
+    "This document describes how to build Python function-based components and use them in your pipeline.\n",
+    "\n",
+    "[component-spec]: https://www.kubeflow.org/docs/pipelines/reference/component-spec/\n",
+    "\n",
+    "## Before you begin\n",
+    "\n",
+    "1. Run the following command to install the Kubeflow Pipelines SDK."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip3 install kfp --upgrade"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. Import the `kfp` and `kfp.components` packages."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp\n",
+    "import kfp.components as comp"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "3. Create an instance of the [`kfp.Client` class][kfp-client].\n",
+    "\n",
+    "[kfp-client]: https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html#kfp.Client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# If you run this command on a Jupyter notebook running on Kubeflow, you can exclude the host parameter.\n",
```

Comment:
Really like the guide @joeliedtke - very awesome!

I'd only suggest to use a couple of shorter comment lines instead of one long one here and in a few places further below where you use code comments. This is to make sure that users don't have to scroll left-to-right often.

<img width="600" alt="image" src="https://user-images.githubusercontent.com/19637339/96640658-65ece200-131b-11eb-92f6-08f39ef55682.png">

Reply:
Good point, done!